### DISCUSSION TITLE
Behaviour on redirect / behaviour on non-Pjax destination

### DISCUSSION THREAD

**Discussion by misog** [PaperStrike/Pjax#341] [Q&A]
Hi, when I submit a form with POST method to logout user (/logout), then there are two HTTP redirections with HTML not containing Pjax (/ and /login). Then, Pjax will visit the logout URL (/logout) **again** with GET. However that will produce 405 Method Not Allowed because the logout URL supports only POST and not GET.

Something similar happens when HTTP session is expired and user clicks. Then there is a redirect to login URL with no Pjax in HTML. However then Pjax tries again to visit the original click (initiator is `document` and not `fetch`). This time, it is a success, because no 405. However, two tries occurs.

Can this be configured? Is the problem that Pjax tries to handle non-Pjax document? Thank you.

**Comment by PaperStrike** [comment]
Pjax simulates the standard browser submissions, so if you found Pjax tries to send an entra request while the standard submission (i.e., disable Pjax and test again) not, please fill a bug report with a minimal reproduction. Thank you.
